sky/spot/controller.py

```diff
@@ -177,13 +182,26 @@ def run(self):
         except exceptions.ResourcesUnavailableError as e:
             logger.error(f'{common_utils.class_fullname(e.__class__)}: '
                          f'{colorama.Fore.RED}{e}{colorama.Style.RESET_ALL}')
```

Comment:
Document self._run() with Raises: (and I think the strategy executor's launch() as well).